File: emJS/euler_02.js
// Encontrar a soma de todos os numeros pares da 
// sequencia de fibonacci até o limite de 4000000


// - Encontrar e somar todos os pares
// - Retornar Sequência de Fibo

function fibo(limite){
  let a = 1;
  let b = 1;
  let c = 0;
  let arr = [];

  while(c < limite){
    c = a + b; // c = 2 // c = 3 // c = 5 //
    a = b;  // a = 1 // a = 2 // a = 3 //
    b = c; // b = 2 // b = 3 // b = 5 // 
    arr.push(c);
  }

  return (arr);
}

function verificaPar(){
  const arr = fibo(4000000);

  let soma = 0;
  for (num of arr){
    if (num % 2 === 0){
      soma += num;
    }    
  }
  console.log(soma);

}

verificaPar()

	

File: emJS/euler_02.py
# Encontra e soma todos os numeros pares da 
# sequencia de fibonacci até o limite de 4000000

def encontra_fib():
	seq = [1,2]
	limite = 4000000
	while (seq[-1] <= limite):
		soma = seq[-1] + seq[-2]
		if (soma < limite):
			seq.append(soma)
		else:
			return seq
	
seq = encontra_fib()

def encontra_pares(seq):
	pares = []
	for indice in seq:
		if (indice % 2 == 0 ):
			pares.append(indice)
	print(pares)
	return pares


pares = encontra_pares(seq)

def soma(pares):
	soma = 0
	for par in pares:
		soma += par
	return soma
	 
soma = soma(pares)
print ("Soma = " + str(soma))

	